src/orion/algo/space/__init__.py

```diff
@@ -67,6 +71,64 @@ def __repr__(self):
         return "..."
 
 
+def _to_snake_case(name: str) -> str:
+    """Transform a class name ``MyClassName`` to snakecase ``my_class_name``"""
+    frags = []
+
+    frag = []
+    for c in name:
+        if c.isupper() and frag:
+            frags.append("".join(frag).lower())
+            frag = []
+
+        frag.append(c)
+
+    if frag:
+        frags.append("".join(frag).lower())
+
+    return "_".join(frags)
+
+
+T = TypeVar("T")
+
+
+class SpaceConverter(Generic[T]):
```

Comment:
I don't really see the point of having a `SpaceConverter`. Is it to force an external library to provide conversion functions functions for the given types of spaces?

Also, this generic signature isn't really respected by the `ToConfigSpace` class, right? Since it doesn't support `Fidelity`-type spaces, it returns `None` in that case, and so it becomes a `SpaceConverter[Optional[HyperParameter]]`!

Reply:
bump

Reply:
> Users can look at the visitor and see clearly all the methods he is expected to override to implement a valid feature.

I guess this answers my first question. However, it's clear from the `fidelity` example that you actually don't want to enforce that every type of dimension has to be supported by subclasses. Therefore, it's not actually an interface, if all its members are optional. Therefore it's only use, as far as I can see, is to be a list of names of methods that "can" be implemented by subclasses.

> NB: Visitor are more than just for the conversion they can be used to implement any new features outside of Dimensions 
Users could implement their own sample/serialization/... using a Visitor.

At this point, why not just let the user code create a function that does what they want to do, e.g. serialize / print / do something for the types of `Dimension` of Orion they want to support?

Reply:
To be fair the fidelity is the only exception. I don't see how the conversions could be usable if they do not support all other type of dimensions. I think the abstract class is the clearest way to express the interface expected for contributors. The only thing needed would be to clarify the exception for the fidelity.